pyscf_ipu/experimental/mesh.py

```diff
@@ -28,13 +28,20 @@ def uniform_mesh(
     axes = [jnp.linspace(-bi, bi, ni) for bi, ni in zip(b, n)]
     mesh = jnp.stack(jnp.meshgrid(*axes, indexing="ij"), axis=-1)
     mesh = mesh.reshape(-1, ndim)
-    return mesh
+    return mesh, axes
 
 
 def electron_density(
     basis: Basis, mesh: FloatNx3, C: Optional[FloatNxN] = None
 ) -> FloatN:
-    C = jnp.eye(basis.num_orbitals) if C is None else C
-    orbitals = basis(mesh) @ C
+    orbitals = molecular_orbitals(basis, mesh, C)
     density = jnp.sum(basis.occupancy * orbitals * orbitals, axis=-1)
     return density
+
+
+def molecular_orbitals(
+    basis: Basis, mesh: FloatNx3, C: Optional[FloatNxN] = None
+) -> FloatN:
+    C = jnp.eye(basis.num_orbitals) if C is None else C
+    orbitals = basis(mesh) @ C
+    return orbitals
```

Comment:
Will this be used somewhere else?  Otherwise I would not pull it out - it forces the reader to jump around in the code.

Reply:
I was planning on using it in the nanoDFT notebook but then realised I would need to #92 implemented to do that.  

I do think it makes sense to have separate API entry points`electron_density` and `molecular_orbitals` since both are useful to have (even if one is implemented in terms of the other).